src/sentry/search/snuba/executors.py

```diff
@@ -156,8 +156,19 @@ def _convert_search_filters(
         """Converts the SearchFilter format into snuba-compatible clauses"""
         converted_filters: list[Optional[Sequence[Any]]] = []
         for search_filter in search_filters or ():
+            conditions, projects_to_filter, group_ids = format_search_filter(
+                search_filter,
+                params={
+                    "organization_id": organization_id,
+                    "project_id": project_ids,
+                    "environment": environments,
+                },
+            )
+            # if no re-formatted conditions, use fallback method
             converted_filters.append(
-                convert_search_filter_to_snuba_query(
+                conditions[0]
+                if conditions
+                else convert_search_filter_to_snuba_query(
```

Comment:
Kind of wonder if this `try format_search_filter else convert_search_filter_to_snuba_query` logic should be encapsulated somewhere. We only do it in two places though so maybe it's fine.

Reply:
probably but this logic is already kind of supposed to be unified. there's a big comment about it in one of the files. I think in general this entire situation needs a refactor/overhaul considering we have a lot of unused code and it's not clear where all of the query codepaths are